fault-proof/src/proposer.rs

```diff
@@ -218,9 +564,14 @@ where
                 .network_prover
                 .prove(&self.prover.range_pk, &sp1_stdin)
                 .compressed()
-                .strategy(FulfillmentStrategy::Reserved)
                 .skip_simulation(false)
```

Comment:
Let's follow upstream as they introduced `self.config.range_cycle_limit` so this `skip_simulation(false)` doesn't work anymore. If we really want to force the proposer to simulate proof generation and estimate `cycle_limit` and `gas_limit`, we should remove `.cycle_limit(self.config.range_cycle_limit)` and `.gas_limit(self.config.range_gas_limit)`. But I think it's enough to have such large default value for `cycle_limit` and `gas_limit` and skip simulation.

```suggestion
                .skip_simulation(true)
```

Reply:
Hey @seolaoh, so previously did we do the simulation to avoid the chance of us using far too many cycles on the proving network?

Reply:
Previously the `cycle_limit` was set as a constant which was much smaller than current default value of `self.config.range_cycle_limit`. So we need to manually change the value when we run into out of cycle_limit error. So we changed it to simulate the execution and estimate the `cycle_limit` dynamically. But it takes quite a long time to simulate that and now we have a much larger value for the config, so I think it's fine to remove the simulation step.